cpp/src/arrow/compute/kernels/scalar_nested.cc

```diff
@@ -165,6 +216,16 @@ void RegisterScalarNested(FunctionRegistry* registry) {
                                          ListValueLength<LargeListType>));
   DCHECK_OK(registry->AddFunction(std::move(list_value_length)));
 
+  auto list_element = std::make_shared<ScalarFunction>("list_element", Arity::Binary(),
+                                                       &list_element_doc);
+  AddListElement<ListType, Int32Type>(list_element.get());
+  AddListElement<ListType, Int64Type>(list_element.get());
```

Comment:
Why index is not any integral type?

Reply:
Could you please share the documentation about "integral types"? I could not find the term in the arrow documentation.
I thought using int32/64 for the index scalar would be the way to go here.

Reply:
@edponce If we want unsigned we get some compiler errors, let me try to make this possible.

Reply:
@edponce I think now we have full support for any integral type for the index, I'll push my changes later, thanks!